storage/src/qmdb/any/unordered/sync.rs

```diff
@@ -97,7 +97,8 @@ where
         .await?;
         // Build the snapshot from the log.
         let snapshot = Index::new(context.with_label("snapshot"), db_config.translator.clone());
-        let db = Self::from_components(range.start, log, snapshot).await?;
+        let db =
+            Self::from_components(range.start, log, snapshot, Location::new_unchecked(0)).await?;
```

Comment:
We're setting the last commit location to 0 here, which may not be correct. This isn't a new bug, exactly -- in the existing code, we leave the last commit location as None. Figuring the last commit location [inside `from_components`](https://github.com/commonwarexyz/monorepo/pull/2507/changes#r2620577195) should fix this.